### ISSUE TITLE
Vercel adapter error in vercel app

### ISSUE THREAD

**Issue by arinzejustin** [sveltejs/kit#8911]
### Describe the bug

<html>
<body>

Start to get this error in vercel build immediately after this recent update that is not more than 1 hour now @sveltejs/adapter-vercel

</body>
</html>

### Reproduction

https://github.com/arinzejustin/cryptflix/dashboard

### Logs

```Shell
TypeError: builder.routes is not iterable
--
18:35:57.663 | at Object.adapt (file:///vercel/path0/dashboard/node_modules/@sveltejs/adapter-vercel/index.js:155:32)
18:35:57.664 | at Object.adapt (file:///vercel/path0/dashboard/node_modules/@sveltejs/adapter-auto/index.js:103:19)
18:35:57.664 | at adapt (file:///vercel/path0/dashboard/node_modules/@sveltejs/adapter-auto/index.js:114:31)
18:35:57.664 | at async adapt (file:///vercel/path0/dashboard/node_modules/@sveltejs/kit/src/core/adapt/index.js:28:2)
18:35:57.664 | at async Object.handler (file:///vercel/path0/dashboard/node_modules/@sveltejs/kit/src/exports/vite/index.js:694:6)
18:35:57.664 | at async PluginDriver.hookParallel (file:///vercel/path0/dashboard/node_modules/rollup/dist/es/shared/rollup.js:23626:17)
18:35:57.664 | at async Object.close (file:///vercel/path0/dashboard/node_modules/rollup/dist/es/shared/rollup.js:24931:13)
18:35:57.664 | at async Promise.all (index 0)
18:35:57.664 | at async build (file:///vercel/path0/dashboard/node_modules/vite/dist/node/chunks/dep-5e7f419b.js:44352:13)
18:35:57.664 | at async CAC.<anonymous> (file:///vercel/path0/dashboard/node_modules/vite/dist/node/cli.js:808:9)
18:35:57.693 | Error: Command "npm run build" exited with 1
```

### System Info

```Shell
Vercel app
```

### Severity

blocking an upgrade

**Comment by dit7ya** [comment]
I can confirm I am also getting the same error on Vercel and locally via `yarn build` as well. I guessing this is due to the fact that `adapter-auto` does not enforce the same semver on the `adapter-vercel` package. (So my `adapter-auto` is at 1.0.0, while it tries to use `adapter-vercel` of 2.0.0).

I will try to post a minimal repro tomorrow.

**Comment by Smirow** [comment]
see https://github.com/sveltejs/kit/issues/8907 ?

**Comment by dit7ya** [comment]
For anybody coming to this thread, you need to update to the following. 

```json
    "@sveltejs/adapter-auto": "^2.0.0",
    "@sveltejs/kit": "^1.5.0",
```

**Comment by Rich-Harris** [comment]
closed via #8915
